src/app/modules/audience/views/engagement/impressions/impressions.component.ts

```diff
@@ -30,7 +30,7 @@ export type funnelData = {
   selector: 'app-engagement-impressions',
   templateUrl: './impressions.component.html',
   styleUrls: ['./impressions.component.scss'],
-  providers: [ImpressionsDataConfig]
+  providers: [ImpressionsDataConfig, ReportService]
```

Comment:
@diamond-darrell  - why do we need to add the ReportService to the providers?

Reply:
Each report should have its own instance of report service, otherwise, it will load only the last report on the page because we have a mechanism that prevents sending multiple duplicating requests, it was fine for pages that have a single report on the page, but for dashboard-type pages, we need to provide it separately